src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/1.sql

```diff
@@ -43,6 +43,9 @@ IF ((SELECT is_ansi_nulls_on FROM sys.databases WHERE database_id = DB_ID()) = 0
     ALTER DATABASE CURRENT SET ANSI_NULLS ON
 END
 
+-- Allow access to versioned rows if Snapshot isolation level is set for a specific transaction
+ALTER DATABASE CURRENT SET ALLOW_SNAPSHOT_ISOLATION ON
+
```

Comment:
As discussed with @brandonpollett, this change may not be reflected (1.sql won't be rerun for existing processes), causing any SQL transactions that specify `IsolationLevel.Snapshot` to throw an exception.

Since it would be better to risk having mismatched count and search results than risk an exception being thrown, I'll revert to using `IsolationLevel.ReadCommitted`. I also created a user story to update the isolation level once schema migration is implemented (see [#AB70783](https://microsofthealth.visualstudio.com/Health/_workitems/edit/70783)).

FYI @deepak-ms @johnstairs

Reply:
Implemented in 8e9b00e.